src/plugins/data/server/field_formats/field_formats_service.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { has } from 'lodash';
+import { FieldFormats } from './field_formats';
+import { IFieldFormatType } from '../../common/field_formats';
+import { IUiSettingsClient } from '../../../../core/server';
+
+import {
+  UrlFormat,
+  StringFormat,
+  NumberFormat,
+  BytesFormat,
+  TruncateFormat,
+  RelativeDateFormat,
+  PercentFormat,
+  IpFormat,
+  DurationFormat,
+  DateNanosFormat,
+  DateFormat,
+  ColorFormat,
+  BoolFormat,
+  SourceFormat,
+  StaticLookupFormat,
+} from '../../common/field_formats';
+
+export class FieldFormatsService {
```

Comment:
Following F2F call.

This service is very similar to `src/plugins/data/public/field_formats_provider/field_formats_service.ts`
I would consider extracting a `common` service or at least some helpers to minimize duplicated code.

Reply:
Agree with you, let's try to reuse one common service for client and server code